homeassistant/components/motioneye/switch.py

```diff
@@ -0,0 +1,120 @@
+"""Switch platform for motionEye."""
+from __future__ import annotations
+
+from types import MappingProxyType
+from typing import Any, Callable
+
+from motioneye_client.client import MotionEyeClient
+from motioneye_client.const import (
+    KEY_MOTION_DETECTION,
+    KEY_MOVIES,
+    KEY_NAME,
+    KEY_STILL_IMAGES,
+    KEY_TEXT_OVERLAY,
+    KEY_UPLOAD_ENABLED,
+    KEY_VIDEO_STREAMING,
+)
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import slugify
+
+from . import MotionEyeEntity, listen_for_new_cameras
+from .const import CONF_CLIENT, CONF_COORDINATOR, DOMAIN, TYPE_MOTIONEYE_SWITCH_BASE
+
+MOTIONEYE_SWITCHES = [
```

Comment:
Most of these settings don't need to be exposed in Home Assistant by default. 

You can control this by setting the `entity_registry_enabled_default` property to `False`

Reply:
Made an attempt to separate into regularly used (or likely to be) vs rarely used, and adjusted enabled/disabled accordingly. Added test.